lib/dataset/bin_detail.cpp

```diff
@@ -28,17 +28,31 @@ Variable make_range(const scipp::index begin, const scipp::index end,
                 dim, CumSumMode::Exclusive);
 }
 
+namespace {
+template <class T> std::tuple<T, Variable> as_edge_view(T &edges) {
+  // The contiguous edges should also be returned along with the view
+  // so that it does not go out of the scope
+  // and kept until the end of the expression where it is called.
+  if (is_bins(edges)) {
+    return std::make_pair(edges, as_subspan_view(edges));
+  }
+  const Dim dim = edges.dims().inner();
+  const auto con_edges = scipp::variable::as_contiguous(edges, dim);
+  return std::make_pair(con_edges, subspan_view(con_edges, dim));
+}
+} // namespace
+
```

Comment:
The helper functions seems to just complicate everything. It is used only once, so why not put the code inline, then the headache about the out-of-scope `con_edges` goes away.

Reply:
I just removed the helper and moved the body of the function but I thought they are around the same since they have to be wrapped in the if ``is_bins`` statement anyways...